perfkitbenchmarker/gcp/gce_network.py

```diff
@@ -30,6 +33,58 @@
 FLAGS = flags.FLAGS
 
 
+class GceFirewallRule(resource.BaseResource):
+  """An object representing a GCE Firewall Rule."""
+
+  def __init__(self, name, project, port):
+    super(GceFirewallRule, self).__init__()
+    self.name = name
+    self.project = project
+    self.port = port
+
+  def __eq__(self, other):
+    """Defines equality to make comparison easy."""
+    return (self.name == other.name and
+            self.port == other.port and
+            self.project == other.project)
+
+  def _Create(self):
+    """Creates the Firewall Rule."""
+    create_cmd = [FLAGS.gcloud_path,
+                  'compute',
+                  'firewall-rules',
+                  'create',
+                  self.name,
+                  '--allow', 'tcp:%d' % self.port, 'udp:%d' % self.port]
+    create_cmd.extend(util.GetDefaultGcloudFlags(self))
+    vm_util.IssueCommand(create_cmd)
+
+  def _Delete(self):
+    """Deletes the Firewall Rule."""
+    delete_cmd = [FLAGS.gcloud_path,
+                  'compute',
+                  'firewall-rules',
+                  'delete',
+                  self.name]
+    delete_cmd.extend(util.GetDefaultGcloudFlags(self))
+    vm_util.IssueCommand(delete_cmd)
+
+  def _Exists(self):
+    """Returns True if the Firewall Rule exists."""
+    describe_cmd = [FLAGS.gcloud_path,
+                    'compute',
+                    'firewall-rules',
+                    'describe',
+                    self.name]
+    describe_cmd.extend(util.GetDefaultGcloudFlags(self))
+    stdout, _, _ = vm_util.IssueCommand(describe_cmd, suppress_warning=True)
+    try:
+      json.loads(stdout)
```

Comment:
Maybe check the return code rather than relying on JSON parsing failure to determine whether the rule exists?